### PR TITLE
Adds Reactive Armor for RD

### LINKED ISSUE #5308: Reactive Teleport Armor
## Description

With the addition of #5279's teleportation function, I think it'd be handy to provide the Research Director with their classic reactive teleport armor, which is a **high-risk item** for traitors. All it does is teleport the wearer to a random spot a short distance away when they are hit with an attack, with a short cooldown before it can do it again.

Adding this item would give at least _something_ to the RD to feel special and important with.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #5308

CL: [New] Adds the Reactive Armor for the Research Director. An armor that teleports its wearer to a random location when getting hit.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
Forgot to update the network manager, going to do that tomorrow

File: UnityProject/Assets/Scripts/Items/Science/Clothing/ReactiveArmor.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	/// <summary>
+	/// A special type of armor that reacts to being hit. Randomly teleporting the wearer to a different location when getting hit.
+	/// </summary>
+	public class ReactiveArmor : MonoBehaviour, IServerInventoryMove
+	{
+		private Pickupable pickupable;
+
+		[SerializeField] private List<DamageType> blacklistedDamageTypes = new List<DamageType> {
+			DamageType.Clone,
+			DamageType.Stamina,
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Science/Clothing/ReactiveArmor.cs:17]: wouldn't it mean it wouldn't trigger on disablers, though we don't have implemented yet?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Science/Clothing/ReactiveArmor.cs:17]: When it's properly implemented, we can worry about that.

**Comment by MaxIsJoe** [comment]
Updated the network manager

File: UnityProject/Assets/Scripts/Items/Science/Clothing/ReactiveArmor.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Science/Clothing/ReactiveArmor.cs:17]: cool

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**